src/widget/flex.rs

```diff
@@ -165,21 +169,28 @@ impl<T: Data> Widget<T> for Flex<T> {
             }
         }
 
-        let total_major = self.direction.major(bc.max);
-        let remaining = total_major - total_non_flex;
+        let total_major = self.direction.major(bc.max());
+        let remaining = (total_major - total_non_flex).max(0.0);
         let flex_sum: f64 = self.children.iter().map(|child| child.params.flex).sum();
 
         // Measure flex children.
         for child in &mut self.children {
             if child.params.flex != 0.0 {
                 let major = remaining * child.params.flex / flex_sum;
+
+                let min_major = if major == std::f64::INFINITY {
```

Comment:
you can also write, `if major.is_infinite() { .. }`.

I might rewrite this as,

```rust
let major = if child.params.flex.is_infinite() {
    0.0 
} else {
    remaining * child.params.flex / flex_sum
};
```

but that's just for the sake of conversation, this certainly works and is clear.